modules/openapi-generator/src/main/resources/swift5/api.mustache

```diff
@@ -157,8 +161,9 @@ extension {{projectName}}API {
     {{/isDeprecated}}
     @available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
     {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}open{{/nonPublicApi}} class func {{operationId}}({{#allParams}}{{paramName}}: {{#isEnum}}{{#isContainer}}{{{dataType}}}{{/isContainer}}{{^isContainer}}{{{datatypeWithEnum}}}_{{operationId}}{{/isContainer}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}}{{^required}}? = nil{{/required}}{{^-last}}, {{/-last}}{{/allParams}}{{#hasParams}}, {{/hasParams}}apiResponseQueue: DispatchQueue = {{projectName}}API.apiResponseQueue) -> AnyPublisher<{{{returnType}}}{{^returnType}}Void{{/returnType}}, Error> {
+        {{#useURLSession}}var dataTask: URLSessionDataTask?{{/useURLSession}}
         return Future<{{{returnType}}}{{^returnType}}Void{{/returnType}}, Error> { promise in
```

Comment:
@4brunu I have one question about this.
Don't we want to wrap this `Future publisher` into a [Deferred publisher](https://developer.apple.com/documentation/combine/deferred)?
So it would work like Rx's `Single` by having it defer its execution until it receives a subscriber.

Reply:
I don't have much experience with `RxSwift` or `Combine`.
Is the current `RxSwift` integration lazy (defer its execution until it receives a subscriber)?

Reply:
@4brunu Yes. RxSwift Observable's only call when you subscribe to the stream.
In Combine publishers are working the same way except Future. We can only achieve the same if we would wrap Future in Deferred.

So currently if we just write this:
``` swift
let newPet = Pet(id:<id>, category: <category>, name: <name>, photoUrls: [])

let addPetPublisher = PetAPI.addPet(body: newPet)
```
The request will be called without subscribing to the stream with `.sink`

Reply:
This is the way it is, because when I introduced support for `Combine` I didn't have much experience with it, and Future looked like a good fit.
I think we can also make `Combine` lazy, because that would make the API more flexible, by only starting the request when the user want's.
What do you think?

Reply:
Yes, I think we should do that 👍

Reply:
Can this behaviour change create any bugs for the users that already implement Combine in their projects?
For example a network request that never happens because of this change?

Reply:
Unfortunately, yes 😕

Reply:
If somebody used it without subscribing to the stream, it won't be called

Reply:
Hum, let's keep that for a different PR then, and keep this one with just the cancellation, to avoid any friction to merge it.